hybrid/decomposers.py

```diff
@@ -1033,14 +1170,39 @@ def make_origin_embeddings(qpu_sampler=None, lattice_type=None,
                                      and target.has_edge(vec_to_lin((2*x+1, 2*y+1, 0, z)),
                                                          vec_to_lin((2*x,2*y+1,0,z)))
                                      })
+            
         else:
             raise ValueError(f'Unsupported qpu_sampler topology {qpu_type} '
                              'for cubic lattice solver')
 
         proposed_source = _make_cubic_lattice(dimensions)
+    elif lattice_type == 'kings':
+        if qpu_type == 'pegasus':
+            vec_to_lin = dnx.pegasus_coordinates(qpu_shape[0]).pegasus_to_linear
+            L = 3*(qpu_shape[0] - 1)
+            dimensions = (L, L)
+            to_chain = _kings_node_to_pegasus_chain
+        elif qpu_type == 'zephyr':
+            if qpu_shape[1] != 4:
+                raise ValueError('Method is suitable for Zephyr[m,t=4] only')
+            vec_to_lin = dnx.zephyr_coordinates(qpu_shape[0], qpu_shape[1]).zephyr_to_linear
+            L = 4 * qpu_shape[0]
+            dimensions = (L, L)
+            to_chain = _squareNNN_node_to_zephyr_chain
+        else:
+            raise ValueError(f'Unsupported qpu_sampler topology {qpu_type} '
+                             'for kings lattice solver')
+        origin_embedding = {(x, y): to_chain(x, y)
+                            for x in range(L) for y in range(L)
+        }  # Defect free, coordinates
```

Comment:
nit: for consistency with [PEP-8/indentation](https://peps.python.org/pep-0008/#indentation), I'd close the brace on the previous line.